ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/RequestHandlerFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.txn.compactor.handler;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.txn.TxnStore;
+import org.apache.hadoop.hive.ql.txn.compactor.FSRemover;
+import org.apache.hadoop.hive.ql.txn.compactor.MetadataCache;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * A factory class to fetch handlers.
+ */
+public class RequestHandlerFactory {
+  private static final RequestHandlerFactory INSTANCE = new RequestHandlerFactory();
+
+  public static RequestHandlerFactory getInstance() {
+    return INSTANCE;
+  }
+
+  /**
+   * Factory class, no need to expose constructor.
+   */
+  private RequestHandlerFactory() {
+  }
+
+  public List<RequestHandler> getHandlers(HiveConf conf, TxnStore txnHandler, MetadataCache metadataCache,
```

Comment:
`getTasks`

Reply:
This gets the handlers, so this method is required.